test/test_utils/ec2.py

```diff
@@ -105,43 +108,43 @@ def get_efa_ec2_instance_type(default, filter_function=lambda x: x, job_type="")
 
 
 def get_ec2_instance_type(
-    default, processor, filter_function=lambda x: x, efa=False, arch_type="", job_type=""
+    default, processor, filter_function=lambda x: x, arch_type="", job_type=""
 ):
     """
-    Get EC2 instance type from associated EC2_[CPU|GPU]_INSTANCE_TYPE env variable, or set it to a default
-    for contexts where the variable is not present (i.e. PR, Nightly, local testing)
+    Get EC2 instance type from associated EC2_[CPU|GPU]_INSTANCE_TYPE env variable, or set it to a
+    default for contexts where the variable is not present (i.e. PR, Nightly, local testing)
 
-    :param default: Default instance type to use - Should never be p3dn
+    :param default: Default instance type to use
     :param processor: "cpu" or "gpu"
-    :param filter_function: filter_function(instance_type_list) A function that takes the list to be generated by
-    the logic of the get_ec2_instance_type function, and filters the list to only produce "acceptable" instances.
-    For example, this can be a function that only returns multi-gpu instance types from a given list of instance types.
+    :param filter_function: filter_function(instance_type_list) A function that takes the list to be
+    generated by the logic of the get_ec2_instance_type function, and filters the list to only
+    produce "acceptable" instances. For example, this can be a function that only returns multi-gpu
+    instance types from a given list of instance types.
 
-    :return: one item list of instance type -- this is used to parametrize tests, and parameter is required to be
-    a list.
+    :return: one item list of instance type -- this is used to parametrize tests, and parameter is
+    required to be a list.
     """
+    if is_pr_context():
+        if get_num_efa_interfaces_for_instance_type(default) and not is_efa_dedicated():
+            return []
+        return [default]
+
     allowed_processors = ("cpu", "gpu", "neuronx", "neuron", "hpu")
     job_type_str = f"_{job_type.upper()}" if job_type else ""
     if processor not in allowed_processors:
         raise RuntimeError(
             f"Aborting EC2 test run. Unrecognized processor type {processor}. "
             f"Please choose from {allowed_processors}"
         )
-    if default in HEAVY_INSTANCE_LIST and not efa:
-        raise RuntimeError(
-            f"Default instance type should never be one of {HEAVY_INSTANCE_LIST}, but it is {default}"
-        )
     instance_type = os.getenv(f"EC2_{processor.upper()}{job_type_str}_INSTANCE_TYPE")
     if arch_type == "graviton":
         instance_type = os.getenv(
             f"EC2_{processor.upper()}_{arch_type.upper()}{job_type_str}_INSTANCE_TYPE"
         )
-    if not instance_type and is_mainline_context():
+    if not instance_type:
```

Comment:
That is true, though I've left it here because I wanted to make it clear to code readers that not specifying a certain instance-type env variable will definitively yield an empty parametrization.

Also I wanted to prevent any missing edge-case handling in the `filter_function` for empty lists from causing it to fail. I do see that we still pass `[instance_type] if instance_type else []` in line 147, below. I can change that to just be `filter_function([instance_type])`. What do you think?

Reply:
You could do that, or `filter_function([instance_type]) if instance_type else []` - either works for me